pkg/osbuild/hmac_stage.go

```diff
@@ -0,0 +1,47 @@
+package osbuild
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+)
+
+type HMACStageOptions struct {
+	Paths     []string `json:"paths"`
+	Algorithm string   `json:"algorithm"`
+}
+
+func (o *HMACStageOptions) isStageOptions() {}
+
+func (o *HMACStageOptions) validate() error {
+	if o == nil {
+		return nil
+	}
+	if len(o.Paths) == 0 {
+		return fmt.Errorf("'paths' is a required property")
+	}
+	if o.Algorithm == "" {
+		return fmt.Errorf("'algorithm' is a required property")
+	}
+
+	algorithms := []string{
+		"sha1",
+		"sha224",
+		"sha256",
+		"sha384",
+		"sha512",
```

Comment:
Nitpick: It may be good to define these values as constants, so that it is clear what are the valid options.